### ISSUE TITLE
Interrupts Intermittent with v2.0.50

### ISSUE THREAD

**Issue by eely22** [bluzDK/bluzDK-firmware#35]
This is the code:

```
#include "application.h"

volatile int intrflagON = 0;
volatile int intrflagOFF = 0;
volatile int intrflagBRIGHT = 0;
volatile int intrflagDIM = 0;
volatile int selectedBrightness = 1;
unsigned long lastTime = 0;
unsigned long lastTime2 = 0;

char buf[6];

void callbackON(){
    intrflagON = 1;
}

void callbackOFF(){
    intrflagOFF = 1;
}

void callbackBRIGHT(){
    intrflagBRIGHT = 1;
}

void callbackDIM(){
    intrflagDIM = 1;
}


void setup() {
  pinMode(D1, INPUT_PULLUP);
    pinMode(D2, INPUT_PULLUP);
    pinMode(D3, INPUT_PULLUP);
    pinMode(D4, INPUT_PULLUP);

//  attachInterrupt(D1, callbackDIM, FALLING);//problem here?
    attachInterrupt(D2, callbackON, FALLING);
    attachInterrupt(D3, callbackOFF, FALLING);
    attachInterrupt(D4, callbackBRIGHT, FALLING);
}//setup

void loop(){
System.sleep(SLEEP_MODE_CPU);
if (Particle.connected()) {
    unsigned long now = millis();
    if ((now - lastTime) >= 1010) { //simple debouncer??
        lastTime = now;

        if (intrflagON == 1)  {
            intrflagON  = 0;
            if (Particle.connected()) {Particle.publish("bedroomONOFF","1",60, PRIVATE);}
        }//if (intrflagON == 1)

        if (intrflagOFF == 1) {
            intrflagOFF = 0;
            if (Particle.connected()) {Particle.publish("bedroomONOFF","0",60, PRIVATE);}
        }//if (intrflagOFF == 1)


        if (intrflagBRIGHT == 1)  {
            intrflagBRIGHT  = 0;
            if (selectedBrightness < 91) {selectedBrightness = selectedBrightness  + 10;} else {selectedBrightness = 100;}
            itoa(selectedBrightness, buf, 10);
            if (Particle.connected()) {Particle.publish("bedroomBRITE", buf,60, PRIVATE);}
        }//if (intrflagON == 1)

        if (intrflagDIM == 1) {
            intrflagDIM = 0;
            if (selectedBrightness > 10) {selectedBrightness = selectedBrightness  - 10;} else {selectedBrightness = 1;}
            itoa(selectedBrightness, buf, 10);
            if (Particle.connected()) {Particle.publish("bedroomBRITE", buf,60, PRIVATE);}
        }//if (intrflagOFF == 1)

    }//if ((now - lastTime) >= 200)
}//if (Particle.connected())

}//loop
```

Basically it publishes messages that control lighting. Worked fine with 1.1.47 with all 4 interrupts as long as I edited the #define GPIOTE_CONFIG_NUM_OF_LOW_POWER_EVENTS 1 to 6
In v2.0.50-beta.2 it won't publish or does so erratically erratically when attachInterrupt(D1, callbackDIM, FALLING) is uncommented (D1 "fires" on bootup, other interrupts unresponsive). BTW I tested D1 alone as an interrupt pin and it failed there too.

**Comment by eely22** [comment]
It does seem that D0 and D1 are not working properly in v2.0.50. This may be related to the TWI driver being active when it shouldn't be.

**Comment by eely22** [comment]
Calling Wire.end() in the setup function, before any other code is executed, can work-around this issue. This must be a problem with the _begin function being called during startup somewhere?

**Comment by eely22** [comment]
Fixed. The issue had to do with the fact that when we merged bluz and bluz-gw into the same repository, we swapped which SPI/TWI interface was available to the user. So the user app used to use SPI1, but now it uses SPI0. 

The issue is, all the devices in the field have a bootloader that still uses SPI1. So when the device booted, the bootloader would activate SPI1 and consume the incorrect pins. 

The simple work-around until a new version is deployed is to add SPI.end() at the beginning of the setup() function. The permanent fix was to do the same thing (turn off SPI1) when the system firmware is initializing.
